python/paddle/fluid/tests/unittests/test_imperative_basic.py

```diff
@@ -640,6 +641,37 @@ def test_append_activation_in_dygraph3(self):
             res2 = fluid.layers.sigmoid(a)
             self.assertTrue(np.array_equal(res1.numpy(), res2.numpy()))
 
+    def test_append_activation_in_dygraph_use_mkldnn(self):
+        a_np = np.random.uniform(-2, 2, (10, 20, 30)).astype(np.float32)
+        helper = LayerHelper(
+            fluid.unique_name.generate("test"),
+            act="relu",
+            use_mkldnn=True,
+            debug=True)
+        func = helper.append_activation
+        with fluid.dygraph.guard():
+            a = fluid.dygraph.to_variable(a_np)
+            res1 = func(a)
+            res2 = fluid.layers.relu(a)
+            self.assertTrue(helper.appended_op.attrs.get('use_mkldnn', False))
+            self.assertTrue(np.array_equal(res1.numpy(), res2.numpy()))
+
+    def test_append_activation_in_dygraph_global_use_mkldnn(self):
+        a_np = np.random.uniform(-2, 2, (10, 20, 30)).astype(np.float32)
+        fluid.set_flags({'FLAGS_use_mkldnn': True})
```

Comment:
@luotao1 Here I have the same question as [here](https://github.com/PaddlePaddle/Paddle/pull/25774#discussion_r462963970), whether setting the global flag can interfere with other tests running in parallel on CI.
Other than that, LGTM.

Reply:
I can't see any option to test global flags in a different way, also other flags are tested this way. After the last commit the time spent with the flag set is minimum.